File: src/main/java/me/m64diamondstar/ingeniamccore/Wands/Cooldowns.java
package me.m64diamondstar.ingeniamccore.Wands;

import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.Map;

public class Cooldowns {

    static Map<String, Long> cooldowns = new HashMap<>();

    public static void addPlayer(Player player, Long op, Long vipp, Long vip, Long visitor){

        if(player.hasPermission("ingeniawands.owner") || player.isOp()) {
            cooldowns.put(player.getName(), System.currentTimeMillis() + (op));
        }

        else if(player.hasPermission("ingenia.vip+")) {
            cooldowns.put(player.getName(), System.currentTimeMillis() + (vipp));
        }

        else if(player.hasPermission("ingenia.vip")) {
            cooldowns.put(player.getName(), System.currentTimeMillis() + (vip));
        }

        else {
            cooldowns.put(player.getName(), System.currentTimeMillis() + (visitor));
        }
    }



    public static boolean isOnCooldown(Player player){
        if(cooldowns.containsKey(player.getName())) {
            if(cooldowns.get(player.getName()) > System.currentTimeMillis()) {

                long timeleft = (cooldowns.get(player.getName()) - System.currentTimeMillis()) / 1000;
                player.sendMessage("§6§lI§e§lngenia §8§l> §cWait §4" + timeleft + " §4Second(s) §cto use this wand again!");

                return true;
            }
        }
        return false;
    }
}

File: src/main/java/me/m64diamondstar/ingeniamccore/wands/Cooldowns.kt
package me.m64diamondstar.ingeniamccore.wands

import me.m64diamondstar.ingeniamccore.general.player.IngeniaPlayer
import me.m64diamondstar.ingeniamccore.utils.messages.MessageLocation
import java.util.HashMap
import org.bukkit.entity.Player
import kotlin.time.Duration.Companion.seconds

object Cooldowns {
    var cooldowns: MutableMap<String, Long> = HashMap()
    fun addPlayer(player: Player, op: Long, vipp: Long, vip: Long, visitor: Long) {
        if (player.isOp || player.hasPermission("ingenia.team")) {
            cooldowns[player.name] = System.currentTimeMillis() + op
        } else if (player.hasPermission("ingenia.vip+")) {
            cooldowns[player.name] = System.currentTimeMillis() + vipp
        } else if (player.hasPermission("ingenia.vip")) {
            cooldowns[player.name] = System.currentTimeMillis() + vip
        } else {
            cooldowns[player.name] = System.currentTimeMillis() + visitor
        }
    }

    @JvmStatic
    fun isOnCooldown(player: Player): Boolean {
        if (cooldowns.containsKey(player.name)) {
            if (cooldowns[player.name]!! > System.currentTimeMillis()) {
                val timeleft = (cooldowns[player.name]!! - System.currentTimeMillis()) / 1000
                val ingeniaPlayer = IngeniaPlayer(player)
                ingeniaPlayer.sendMessage("&cThis wand is on cooldown for ${timeleft.seconds}.", MessageLocation.HOTBAR)
                return true
            }
        }
        return false
    }
}